imports/plugins/core/ui-tagnav/client/containers/tagNavContainer.js

```diff
@@ -0,0 +1,367 @@
+import debounce from "lodash/debounce";
+import update from "react/lib/update";
+import React, { Component, PropTypes } from "react";
+import { Reaction } from "/client/api";
+import TagNav from "../components/tagNav";
+import { TagHelpers } from "/imports/plugins/core/ui-tagnav/client/helpers";
+
+const styles = {
+  editContainerItem: {
+    display: "flex",
+    marginLeft: 5
+  }
+};
+
+const NavbarStates = {
+  Orientation: "stateNavbarOrientation",
+  Position: "stateNavbarPosition",
+  Anchor: "stateNavbarAnchor",
+  Visible: "stateNavbarVisible"
+};
+
+const NavbarOrientation = {
+  Vertical: "vertical",
+  Horizontal: "horizontal"
+};
+
+const NavbarVisibility = {
+  Shown: "shown",
+  Hidden: "hidden"
+};
+
+const NavbarPosition = {
+  Static: "static",
+  Fixed: "fixed"
+};
+
+const NavbarAnchor = {
+  Top: "top",
+  Right: "right",
+  Bottom: "bottom",
+  Left: "left",
+  None: "inline"
+};
+
+const TagNavHelpers = {
+  onTagCreate(tagName, parentTag) {
+    TagHelpers.createTag(tagName, undefined, parentTag);
+  },
+  onTagRemove(tag, parentTag) {
+    TagHelpers.removeTag(tag, parentTag);
+  },
+  onTagSort(tagIds, parentTag) {
+    TagHelpers.sortTags(tagIds, parentTag);
+  },
+  onTagDragAdd(movedTagId, toListId, toIndex, ofList) {
+    TagHelpers.moveTagToNewParent(movedTagId, toListId, toIndex, ofList);
+  },
+  onUpdateTag(tagId, tagName, parentTagId) {
+    TagHelpers.updateTag(tagId, tagName, parentTagId);
+  },
+  isMobile() {
+    return window.matchMedia("(max-width: 991px)").matches;
+  },
+  tagById(tagId, tags) {
+    return _.find(tags, (tag) => tag._id === tagId);
+  },
+  updateSuggestions(suggestion, excludeTagsObj) {
+    return TagHelpers.updateSuggestions(suggestion, excludeTagsObj);
+  },
+  hasSubTags(tagId, tags) {
+    const foundTag = this.tagById(tagId, tags);
+
+    if (foundTag) {
+      if (Array.isArray(foundTag.relatedTagIds) && foundTag.relatedTagIds.length) {
+        return true;
+      }
+    }
+    return false;
+  }
+};
+
+let editable = false;
+let selectedTag = null;
+
+class TagNavContainer extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      attachedBodyListener: false,
+      editable: editable,
+      tagIds: props.tagIds || [],
+      tagsByKey: props.tagsByKey || {},
+      selectedTag: selectedTag,
+      suggestions: [],
+      [NavbarStates.Visible]: false,
+      [NavbarStates.Visibile]: NavbarVisibility.Hidden,
+      newTag: {
+        name: ""
+      }
+    };
+  }
+
+  componentDidMount() {
+    $(window).on("resize", this.onWindowResize).trigger("resize");
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
I dont think this is necessary. Instead of passing props into state then using that state, just use them as props.